7.6.3 adds dependency on node ^20.11.0

# Problem
As of the 7.6.3 release my build stopped working

## Issue
As of 7.6.3 when building my project npm now gives this error
```
npm ERR! notsup Required: {"node":"^20.11.0 || >=21.2.0"} 
npm ERR! notsup Actual:   {"npm":"10.5.0","node":"v18.20.1"} 
```

The change was introduced in https://github.com/focus-trap/focus-trap/pull/1336.

```
 "engines": {
    "node": "^20.11.0 || >=21.2.0"
  },
```

Not sure what the solution is but I feel like this was not expected.  Not sure if the solution is to bump the minor version and delete the 7.6.3 release.

--- a/.changeset/thirty-baboons-push.md
+++ b/.changeset/thirty-baboons-push.md
@@ -1,0 +1,5 @@
+---
+'focus-trap': patch
+---
+
+Remove engine requirement that crept into `package.json`. Sorry about that! ([#1343](https://github.com/focus-trap/focus-trap/issues/1343))

--- a/eslint.config.mjs
+++ b/eslint.config.mjs
@@ -4,6 +4,8 @@
 
 /* eslint-env node */
 
+import url from 'node:url';
+import path from 'node:path';
 import js from '@eslint/js';
 import globals from 'globals';
 import babel from '@babel/eslint-plugin';
@@ -17,9 +19,11 @@ import cypress from 'eslint-plugin-cypress';
 import importPlugin from 'eslint-plugin-import';
 import testingLibrary from 'eslint-plugin-testing-library';
 
+const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
+
 const ecmaVersion = 'latest';
 const impliedStrict = true;
-const tsconfigRootDir = import.meta.dirname;
+const tsconfigRootDir = __dirname;
 
 //
 // Plugins

--- a/package.json
+++ b/package.json
@@ -2,9 +2,6 @@
   "name": "focus-trap",
   "version": "7.6.3",
   "description": "Trap focus within a DOM node.",
-  "engines": {
-    "node": "^20.11.0 || >=21.2.0"
-  },
   "main": "dist/focus-trap.js",
   "module": "dist/focus-trap.esm.js",
   "types": "index.d.ts",

